docs/guides/snapshot/process.md

```diff
@@ -74,32 +74,30 @@ minikube    demo         mypod    mypod        sleep    584294
 We can also get the information in JSON format, by passing the `-o json` flag.
 
 ```bash
-$ kubectl gadget snapshot process -n demo -o json
-[
-  {
-    "type": "normal",
-    "node": "minikube",
-    "namespace": "demo",
-    "pod": "mypod",
-    "container": "mypod",
-    "tgid": 582294,
-    "pid": 582294,
-    "comm": "nginx",
-    "mntns": 4026532507
-  },
-  {
-    "type": "normal",
-    "node": "minikube",
-    "namespace": "demo",
-    "pod": "mypod",
-    "container": "mypod",
-    "tgid": 582333,
-    "pid": 582333,
-    "comm": "nginx",
-    "mntns": 4026532507
-  },
-  ...
-]
+$ kubectl gadget snapshot process -n demo -o json | jq
+{
+  "type": "normal",
+  "node": "minikube",
+  "namespace": "demo",
+  "pod": "mypod",
+  "container": "mypod",
+  "tgid": 582294,
+  "pid": 582294,
+  "comm": "nginx",
+  "mntns": 4026532507
+}
+{
+  "type": "normal",
+  "node": "minikube",
+  "namespace": "demo",
+  "pod": "mypod",
+  "container": "mypod",
+  "tgid": 582333,
+  "pid": 582333,
+  "comm": "nginx",
+  "mntns": 4026532507
+}
```

Comment:
TL;DR: I think this result is wrong and we should keep the previous one.

We need to understand what it means to have json output on the different gadget categories. For gadgets that produce events, like trace, top, audit, the full output of the tool is not a json array of events, instead each single line is a json itself that represents the event. On the other hand, gadgets in the snapshot category produce a full json array as result. 

I think it's useful to have snapshotters to produce a valid json output, I'm not sure what to do for tracers and similar, as they don't have all the output available at one hence it's not possible to print the event in a single call. 
Maybe we can print a `[` at the beginning, separate lines by `,`, and print a `]` before exiting? (Or maybe it's too complicated and we shouldn't worry about it?).  

@alban do you have any opinions on this?

Reply:
Even if I agree that having the same behavior would be nice, I also think that different categories can have different behaviors:

> $ kubectl gadget trace tcp -o json | myapp

In this case myapp should be prepared to process one event at the time (currently each new line represents a new event). 

> $ kubectl gadget snapshot socket -o json | myapp

In this case it's expected to received all the information on a single step, so probably myapp would read all and parse as a single json structure. 

I agree adding `[` to the tracers is too much, but I think snapshotters should keep the current behavior.

Reply:
> TL;DR: I think this result is wrong and we should keep the previous one.

I agree with that.
* Snapshotters: with should be able to use `| jq`, so we need valid json.
* Streams cannot be parsed as valid json and adding `[` would not help because a partial json document would still not be parsable. I think the separation by line was good.

Reply:
Make sense. I love uniformity 😝, but it is also true that one of the reasons for dividing the gadgets into categories is that in some cases, like this one, we want to have different formats between categories while keeping uniformity within each category. I will revert commits for the snapshot category.

Reply:
BTW, `jq` can still be used even if each line is a JSON by itself (like in the case of streams):

```bash
$ printf '{"a": 1}\n{"a": 2}\n'|jq .a
1
2
```